crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -415,7 +416,15 @@ fn execute_syscall(
     }
 
     let gas_counter = buffer.next_usize().unwrap();
+
     let contract_address = buffer.next_felt252().unwrap().into_owned();
+    let contract_address = ContractAddress(
+        PatriciaKey::try_from(
+            StarkFelt::new(contract_address.to_be_bytes()).expect("Felt conversion failed"),
+        )
+        .expect("PatriciaKey failed"),
+    );
```

Comment:
I think it would make sense to introduce helpers handling this conversions. This can be done in separate PR.

Reply:
#489